lib/round.rb

```diff
@@ -0,0 +1,52 @@
+class Round
+    attr_reader :deck, :turns, :cards, :number_correct
+    
+    def initialize (deck)
+        @deck = deck
+        @cards = cards
+        @turns = []
+        @number_correct = 0
+        @correct_turns = @correct_turns
+        @category_turns = @category_turns
+
+    end
+
+    def current_card
+        deck.cards[0]
+    end
+
+    def take_turn(guess)
+        turn = Turn.new(guess, current_card)
+        @turns << turn
+        if guess == current_card.answer
+            @number_correct += 1 
+        end
+        deck.cards.rotate!
+        return turn
+    end
+
+    def number_correct_by_category(category)
+
+        @correct_turns = turns.select do |turn|
```

Comment:
This method works just fine as written but just as an alternate to this method, you could also just chain on the `count` method directly to the block like so:

```ruby
def number_correct_by_category(category)
  turns.select do |turn|
    turn.card.category == category && turn.correct?
  end.count # <====
end
```

We can do this because the `select` enumerable returns and array of selected items, and then we are just calling `count` on that returned array.